src/instructlab/sdg/default_flows.py

```diff
@@ -45,29 +44,25 @@ class _SimpleFlow(Flow):
     def get_flow(self) -> list:
         return [
             {
-                "block_type": IterBlock,
+                "block_type": LLMBlock,
                 "block_config": {
                     "block_name": "",  # must be set by subclass
-                    "num_iters": self.num_iters,
-                    "block_type": LLMBlock,
-                    "block_kwargs": {
-                        "block_name": "",  # must be set by subclass
-                        "config_path": "",  # must be set by subclass
-                        "client": self.client,
-                        "model_id": self.model_id,
-                        "model_prompt": _get_model_prompt(self.model_family),
-                        "output_cols": ["output"],
-                        "batch_kwargs": {
-                            "num_procs": 8,
-                            "batched": self.batched,
-                        },
-                    },
-                    "gen_kwargs": {
-                        "max_tokens": 2048,
-                        "temperature": 0.7,
+                    "config_path": "",  # must be set by subclass
+                    "client": self.client,
+                    "model_id": self.model_id,
+                    "model_prompt": _get_model_prompt(self.model_family),
+                    "output_cols": ["output"],
+                    "batch_kwargs": {
+                        "num_procs": 8,
+                        "batched": self.batched,
                     },
-                    "drop_duplicates": ["output"],
                 },
+                "gen_kwargs": {
+                    "max_tokens": 2048,
+                    "temperature": 0.7,
+                    "n": 1
```

Comment:
Should this be `self.num_iters`?

Reply:
No this is a vllm specific parameter - we are replacing the IterBlock with this. This parameter essentialy controls the number of return sequences

Reply:
No, these parameters get passed to the openai client directly as kwargs. Ref to openai's `n` parameter https://platform.openai.com/docs/api-reference/completions/create#completions-create-n. In theory, openai's functionality of `n` is the same as what we were doing with `num_iters`.

Reply:
Running `ilab data generate` with this branch gives:

```
  File "/home/markmc/sdg/src/instructlab/sdg/generate_data.py", line 257, in generate_data
    sdg_knowledge, sdg_freeform_skill, sdg_grounded_skill = _sdg_init(
                                                            ^^^^^^^^^^
  File "/home/markmc/sdg/src/instructlab/sdg/generate_data.py", line 144, in _sdg_init
    [
  File "/home/markmc/sdg/src/instructlab/sdg/generate_data.py", line 146, in <listcomp>
    flow_type(
TypeError: Flow.__init__() takes from 4 to 5 positional arguments but 6 were given
```

(after I fix the iterblock import error)

What should the `num_instructions_to_generate` parameter to `generate_data()` be used for?

Reply:
@shivchander or @oindrillac any thoughts here? As far as I know the `generate_data.py` file still contains older sdg code right?

Reply:
@markmc For testing it out you can use the `test_*` scripts

```
cd sdg
time python3 scripts/test_grounded_skills.py
```

Reply:
`ilab` now uses this library via the `generate_data()` API. It doesn't matter if the test scripts work - what's important is that `ilab data generate` is broken

Reply:
Yeah, it seems like `generate_data.py` has been updated to use the newer pipeline structure. So I guess I can go ahead and fix the `num_iter` in `generate_data()` API call. Just want to get confirmation from @oindrillac / @aakankshaduggal

Reply:
I think the point is that `num_iters` was a constructor arg to the `Flow` class before and it's not clear why that's changed. Is it the intention to change this from being parameterized to being 1? If it's always 1 then there's no need to specify `n` at all. If it stays as a param to the Flow class then it might be best to rename it since `num_iters` isn't really correct.. should be something more like `num_parallel_outputs_per_input`

Reply:
Thanks @njhill 

Again the question @npalaska is ... how should the `num_instructions_to_generate` parameter to `generate_data()` be implemented?

(Is it not by specifying `n`?)

Reply:
Alright, after discussing with @shivchander, we think that the client `num_instructions_to_generate` can be captured by openai's `n` parameter.  Currently, the `num_samples` are hardcoded in the `default_flows.py` which is overwriting the `num_instructions_to_generate` /`num_iters` passed from `generate_data()`. It should be fixed in the next commit.

Reply:
does that `n` parameter imply that the server supports batching?  llama-cpp does not